Pretty, human-readable output for make logs

Part of the mission-observability epic (#2361).

## Problem

`make logs` tails the raw log files, so a working mission looks like:

```
[cli] assistant — tool_use: Edit
[cli] tool_result toolu_015i88
[cli] system: thinking_tokens
[cli] assistant — thinking
[cli] assistant — text: Now run.py — inject PYTEST_ADDOPTS and add post-mission sweep.
```

Informative, but hard to follow for a human watching Kōan work.

## Proposal

A **display-side formatter** piped after `tail -F` in the `make logs` target (e.g. `koan/app/log_fmt.py`, stdin→stdout):

- `[cli] assistant — thinking` and `[cli] system: thinking_tokens` → a single dim `•` (collapse consecutive repeats)
- `[cli] assistant — text: <preview>` → `🧠 <preview>`
- `[cli] assistant — tool_use: <Name>` → `🔧 <Name>` (optionally per-tool icons: ✏️ Edit, 📖 Read, 💻 Bash…)
- `[cli] tool_result …` → hidden or dim `↩`
- Non-`[cli]` lines (run.py lifecycle, awake.py, pending.md) pass through untouched
- Colors only when stdout is a TTY

Add a `make logs raw=1` escape hatch that skips the filter.

## Hard constraint

Purely presentational. The log **files** and the canonical `[cli]` format emitted by `_summarize_stream_event()` (`koan/app/provider/__init__.py`) are unchanged — those lines are parsed by `run.py`, `quota_handler.py`, `jira_outcome_publish.py`, and `claude_step.py`.

## Acceptance

- `make logs` shows the formatted stream; `make logs raw=1` shows today's output
- `tail -F ... | log_fmt` never crashes on partial/garbled lines (passes unknown lines through)
- Unit tests on the formatter with representative `[cli]` fixtures

does not work as expected: 

````
$ make logs
→ Watching Kōan logs + live progress (Ctrl-C to stop watching — Kōan keeps running)
/bin/sh: 5: /app/.venv/bin/python3: not found
make: *** [Makefile:295: logs] Error 127
````

@sukria-koan0 fix

--- a/Makefile
+++ b/Makefile
@@ -298,8 +298,14 @@ logs:
 	@if [ "$(raw)" = "1" ]; then \
 		tail -F logs/run.log logs/awake.log logs/ollama.log instance/journal/pending.md 2>/dev/null; \
 	else \
-		tail -F logs/run.log logs/awake.log logs/ollama.log instance/journal/pending.md 2>/dev/null \
-			| ( cd koan && PYTHONPATH=. $(PYTHON_ABS) -m app.log_fmt ); \
+		fmt_py="$(PYTHON_ABS)"; \
+		[ -x "$$fmt_py" ] || fmt_py="$$(command -v python3 || command -v python)"; \
+		if [ -z "$$fmt_py" ]; then \
+			tail -F logs/run.log logs/awake.log logs/ollama.log instance/journal/pending.md 2>/dev/null; \
+		else \
+			tail -F logs/run.log logs/awake.log logs/ollama.log instance/journal/pending.md 2>/dev/null \
+				| ( cd koan && PYTHONPATH=. "$$fmt_py" -m app.log_fmt ); \
+		fi; \
 	fi
 
 install:

--- a/docs/operations/log-formatting.md
+++ b/docs/operations/log-formatting.md
@@ -21,7 +21,7 @@ presentational — the log **files** and the `[cli]` grammar emitted by
 | Raw `[cli]` line                    | Shown as        |
 |-------------------------------------|-----------------|
 | `assistant — thinking`, `system: thinking_tokens` | dim `•` (consecutive collapse) |
-| `assistant — text: <preview>`       | `🧠 <preview>`   |
+| `assistant — text: <preview>`       | `🧠 <preview>` (preview skips leading code fences / bare brackets, e.g. ```` ```json ```` → first real line) |
 | `assistant — tool_use: Edit`        | `✏️ Edit` (per-tool icons; `🔧` default) |
 | `tool_result …`                     | dim `↩`         |
 | `tool_result … (error)`             | `❌ tool error` (high-signal; never collapses) |

--- a/koan/app/provider/__init__.py
+++ b/koan/app/provider/__init__.py
@@ -727,10 +727,25 @@ def _content_text(content: Any) -> str:
     return ""
 
 
+# Leading lines that carry no readable preview on their own: code-fence
+# markers (```/```json), bare structural punctuation ({, }, [], ...), and
+# horizontal rules. Skipped when picking a preview so a text block that opens
+# with ```json renders its actual content instead of "🧠 {".
+_LOW_SIGNAL_PREVIEW = re.compile(r"^(?:`{3,}[\w+-]*|[-{}\[\]()]+|-{3,})$")
+
+
 def _first_line(value: Any, limit: int = 80) -> str:
-    """First line of *value* as a bounded preview string ('' when empty)."""
-    lines = str(value or "").strip().splitlines()
-    return lines[0][:limit] if lines else ""
+    """First meaningful line of *value* as a bounded preview ('' when empty).
+
+    Skips leading blank lines and low-signal markers (code fences, bare
+    brackets, rules) so a block that opens with ```json or ``{`` surfaces its
+    real content instead of the marker.
+    """
+    for line in str(value or "").splitlines():
+        stripped = line.strip()
+        if stripped and not _LOW_SIGNAL_PREVIEW.match(stripped):
+            return stripped[:limit]
+    return ""
 
 
 def _summarize_stream_event(event: Dict[str, Any]) -> str:
@@ -762,7 +777,7 @@ def _summarize_stream_event(event: Dict[str, Any]) -> str:
             elif btype == "text":
                 text = (block.get("text") or "").strip()
                 if text:
-                    preview = text.splitlines()[0][:80]
+                    preview = _first_line(text) or text.splitlines()[0][:80]
                     parts.append(f"text: {preview}")
                 else:
                     parts.append("text")
@@ -858,7 +873,8 @@ def _summarize_stream_event(event: Dict[str, Any]) -> str:
         if item_type == "message" or item.get("role") == "assistant":
             text = _content_text(item.get("content")).strip()
             if text:
-                return f"[cli] assistant — text: {text.splitlines()[0][:80]}"
+                preview = _first_line(text) or text.splitlines()[0][:80]
+                return f"[cli] assistant — text: {preview}"
             return "[cli] assistant — message"
         if item_type:
             suffix = f" ({status})" if status else ""

